kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/CrunchDatasets.java

```diff
@@ -191,13 +191,43 @@ public static Target asTarget(URI uri) {
   public static <E> PCollection<E> partition(PCollection<E> collection,
                                              View<E> view,
                                              int numWriters) {
+    return partition(collection, view, numWriters, 1);
+  }
+
+  /**
+   * Partitions {@code collection} to be stored efficiently in {@code View}.
+   * <p>
+   * This restructures the parallel collection so that all of the entities that
+   * will be stored in a given partition will be evenly distributed across a specified
+   * {@code numPartitionWriters}.
+   * <p>
+   * If the dataset is not partitioned, then this will structure all of the
+   * entities to produce a number of files equal to {@code numWriters}.
+   *
+   * @param collection a collection of entities
+   * @param view a {@link View} of a dataset to partition the collection for
+   * @param numWriters the number of writers that should be used
+   * @param numPartitionWriters the number of writers data for a single partition will be distributed across
+   * @param <E> the type of entities in the collection and underlying dataset
+   * @return an equivalent collection of entities partitioned for the view
+   * @see #partition(PCollection, View)
+   *
+   * @since 1.1.0
+   */
+  public static <E> PCollection<E> partition(PCollection<E> collection,
+                                             View<E> view,
+                                             int numWriters, int numPartitionWriters) {
+
+    //ensure the number of writers is honored whether it is per partition or total.
+    int writers = Math.max(numPartitionWriters, numWriters);
```

Comment:
I think these should be separate configuration settings unless the dataset is unpartitioned. I'm fine taking the max of both for that case, but otherwise `writers` should be `numWriters`. So in the `isPartitioned` block, I think you should use `numWriters`, and use `writers` in the else block. Another alternative is to allow `numPartitionWriters` to override `numWriters` or just make keys with only the writer random number so the behavior is the same across both.